test/tests/functional/pbs_ralter.py

```diff
@@ -0,0 +1,1007 @@
+            offset2, duration, select="2:ncpus=4")
+        self.submit_and_confirm_reservation(offset1, duration,
+                                            select="2:ncpus=4",
+                                            ExpectSuccess=0)
+        self.alter_a_reservation(rid1, start1, end1, shift, alter_e=True,
+                                 ExpectSuccess=3)
+        self.alter_a_reservation(rid1, start1, end1, shift, alter_e=True,
+                                 ExpectSuccess=3, interactive=5, sequence=2)
+
+    def test_conflict_two_standing_resvs(self):
+        """
+        This test confirms that a standing reservation's end time cannot be
+        moved back if there is a conflicting reservation and all the nodes
+        in the complex are busy.
+
+        Two back to back standing reservations are submitted that use all the
+        nodes in the complex to test this.
+        """
+        duration = 120
+        shift = 10
+        offset1 = 60
+        offset2 = 180
+
+        rid1, start1, end1 = self.submit_and_confirm_reservation(
+            offset1, duration, select="2:ncpus=4", standing=True)
+        rid2, start2, end2 = self.submit_and_confirm_reservation(
+            offset2, duration, select="2:ncpus=4", standing=True)
+        self.submit_and_confirm_reservation(offset1, duration,
+                                            select="2:ncpus=4",
+                                            ExpectSuccess=0, standing=True)
+        self.alter_a_reservation(rid1, start1, end1, shift, alter_e=True,
+                                 ExpectSuccess=3)
+        self.alter_a_reservation(rid1, start1, end1, shift, alter_e=True,
+                                 ExpectSuccess=3, interactive=5, sequence=2)
+
+    def test_check_alternate_nodes_advance_resv(self):
+        """
+        This test confirms that an advance reservation's end time can be
+        moved back even if there is a conflicting reservation but there are
+        nodes in the complex that satisfy the resource requirements of the
+        reservation.
+
+        Two back to back advance reservations are submitted that use the
+        same nodes in the complex to test this.
+        """
+        duration = 120
+        shift = 10
+        offset1 = 60
+        offset2 = 180
+
+        rid1, start1, end1 = self.submit_and_confirm_reservation(
+            offset1, duration, select="1:ncpus=4")
+
+        self.server.status(RESV, 'resv_nodes', id=rid1)
+        resv_node = self.server.reservations[rid1].get_vnodes()[0]
+
+        select = "1:vnode=" + resv_node + ":ncpus=4"
+        rid2, start2, end2 = self.submit_and_confirm_reservation(
+            offset2, duration, select=select)
+
+        attrs = {'resv_nodes': (MATCH_RE, re.escape(resv_node))}
+        self.server.expect(RESV, attrs, id=rid2)
+
+        attrs = {'resources_available.ncpus': (GT, 0)}
+        vnode_dict = self.server.filter(NODE, attrs)
+        vnode_list = vnode_dict.values()[0]
```

Comment:
yeah, but I like the natural vnode to have 0 cpus. So, I have not made any changes here.